app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -112,7 +112,11 @@ def task_iteration(input)
       end
     rescue => error
       @errored_element = input
-      raise error
+      if @task.exception_handlers.keys.include?(error.class)
```

Comment:
Do we have an option to split it into two `rescue`, like:
```ruby
    rescue *@task.exception_handlers.keys => error
      @errored_element = input
      @task.exception_handlers[error.class].call(error)
    rescue => error
      @errored_element = input
      raise error
```

`@errored_element = input` has to be duplicated so not sure if it's much cleaner. So totally up to you